### PR TITLE
Bump cert-manager API version to v1

### PR DESCRIPTION
**Pull request by nak3**
As per title, this patch bumps cert-manager API version to v1. 
v1alpha2 will be deprecated soon.

Fix https://github.com/knative-sandbox/net-certmanager/issues/219

### CODE REVIEW

File: pkg/reconciler/certificate/certificate.go [hunk 1]
```diff
@@ -150,7 +150,7 @@ func (c *Reconciler) reconcileCMCertificate(ctx context.Context, knCert *v1alpha
 	} else if !equality.Semantic.DeepEqual(cmCert.Spec, desired.Spec) {
 		copy := cmCert.DeepCopy()
 		copy.Spec = desired.Spec
-		updated, err := c.certManagerClient.CertmanagerV1alpha2().Certificates(copy.Namespace).Update(ctx, copy, metav1.UpdateOptions{})
+		updated, err := c.certManagerClient.CertmanagerV1().Certificates(copy.Namespace).Update(ctx, copy, metav1.UpdateOptions{})
```
Comment (markusthoemmes) [inline pkg/reconciler/certificate/certificate.go:153]: Does this work as "expected" for existing certificates? I.e. do they get replaced?

Comment (nak3) [reply pkg/reconciler/certificate/certificate.go:153]: As cert-manager converts the API version from `v1alpha2` to `v1` by cert-manager's [CustomResourceConversionWebhook](https://cert-manager.io/docs/concepts/webhook/), existing certificates.cert-manager.io is already `v1`.
I confirmed it with serving repo's k8s dump, for example [k8s.dump-e2e-tests.sh.txt](https://storage.googleapis.com/knative-prow/pr-logs/pull/knative_serving/11322/pull-knative-serving-https/1396284336117911552/artifacts/5a4c4c7b-bb6a-11eb-9519-8ab7a5ae2478/k8s.dump-e2e-tests.sh.txt) [1].

Actually I wondered why current code is working fine :thinking: 

[1] current certificates.cert-manager.io is already `v1` not `v1alpha2`.
```
>>> certificates.cert-manager.io (3 objects)
>>> Listing
NAMESPACE       NAME                                         READY   SECRET                                       AGE
serving-tests   route-4b65472c-54df-40e7-9895-ff1eb3034a7b   True    route-4b65472c-54df-40e7-9895-ff1eb3034a7b   3m26s
serving-tests   route-763e4fcd-251f-4c93-8715-6e25a1e5fb9a   True    route-763e4fcd-251f-4c93-8715-6e25a1e5fb9a   2m28s
serving-tests   route-d214f424-39ab-4ec0-9e6d-836cef3454f8   True    route-d214f424-39ab-4ec0-9e6d-836cef3454f8   68s
>>> Details
apiVersion: v1
items:
- apiVersion: cert-manager.io/v1
  kind: Certificate
```

Comment (ZhiminXiang) [reply pkg/reconciler/certificate/certificate.go:153]: I confirmed that the current certificate and other types of CRs are v1. I think cert-manager did the auto conversion underneath.

**Comment by ZhiminXiang** [comment]
/lgtm

Thanks for working on this, @nak3 
Currently our public doc still uses v1alpha2 API. Once this PR is checked in, we need to update the doc to v1 to keep the consistency.

**Review by ZhiminXiang — APPROVED** [review]

**Comment by nak3** [comment]
The doc was already updated by https://github.com/knative/docs/pull/3670.

**Pull request merged**